src/components/tl/ucp/allgather/allgather_sparbit.c

```diff
@@ -123,20 +123,29 @@ ucc_status_t ucc_tl_ucp_allgather_sparbit_start(ucc_coll_task_t *coll_task)
     ucc_rank_t         tsize     = UCC_TL_TEAM_SIZE(team);
     size_t             data_size = (count / tsize) * ucc_dt_size(dt);
     ucc_status_t       status;
-
+    int                use_loopback = UCC_TL_UCP_TEAM_LIB(team)->cfg.allgather_use_loopback;
+    
     UCC_TL_UCP_PROFILE_REQUEST_EVENT(coll_task, "ucp_allgather_sparbit_start",
                                      0);
     ucc_tl_ucp_task_reset(task, UCC_INPROGRESS);
     task->allgather_sparbit.i             = 0; // setup iteration
     task->allgather_sparbit.data_expected = 1;
 
     if (!UCC_IS_INPLACE(TASK_ARGS(task))) {
-        status = ucc_mc_memcpy(PTR_OFFSET(rbuf, data_size * trank), sbuf,
+        if (!use_loopback) {
+            status = ucc_mc_memcpy(PTR_OFFSET(rbuf, data_size * trank), sbuf,
                                data_size, rmem, smem);
-        if (ucc_unlikely(UCC_OK != status)) {
-            return status;
+            if (ucc_unlikely(UCC_OK != status)) {
+                return status;
+            }
+        } else {
+            /* Loopback */
+            UCPCHECK_GOTO(ucc_tl_ucp_send_nb(sbuf, data_size, smem, trank, team, task),task, out);
+            UCPCHECK_GOTO(ucc_tl_ucp_recv_nb(PTR_OFFSET(rbuf, data_size * trank), data_size, rmem, trank, team, task),task, out);
```

Comment:
this whole code block is duplicated in neighbor, ring and sparbit. It should be refactored into a helper function and called there.

Ideally, this could also be refactored with the call from knomial, with an additional control about whether to use `ucc_mc_memcpy` or the executor.